python/GafferUI/CompoundEditor.py

```diff
@@ -998,3 +1040,126 @@ def __getSharedDragWidget( self ) :
+
+	if widgetScreen == QtWidgets.QApplication.primaryScreen():
+		widgetScreenNumber = -1
+
+	screenGeom = widgetScreen.availableGeometry()
+	screenW = float( screenGeom.width() )
+	screenH = float( screenGeom.height() )
+
+	# Ideally we'd use frameGeometry for portability, but have seen a variety
+	# of issues where the window has been made visible, but doesn't yet
+	# have a frame, as we rely on asking for the margins (as you can't
+	# setFrameGeometry) we end up with unreliable window placement. As such
+	# we compromise and store the widgets geometry instead.
+	windowGeom = qWidget.normalGeometry()
+
+	x = ( windowGeom.x() - screenGeom.x() ) / screenW
+	y = 1.0 - ( ( windowGeom.y() - screenGeom.y() ) / screenH )
+	w = windowGeom.width() / screenW
+	h = windowGeom.height() / screenH
+
+	return {
+		"screen" :  widgetScreenNumber,
+		"fullScreen" : gafferWindow.getFullScreen(),
+		"maximized" : bool(window.windowState() & QtCore.Qt.WindowMaximized),
+		# As we're bottom-left not top-left the y values are the other way around
+		"bound" : imath.Box2f( imath.V2f( x, y - h ), imath.V2f( x + w, y ) )
+	}
+
+def _imathRepr( data ) :
+
+	# Neither IECore.repr or repr include the module name
+	s = IECore.repr( data )
+	for ss in ( 'Box2f', 'V2f' ) :
+		s = s.replace( ss, "imath.%s" % ss )
+	return s
+
+def _restorePreferredBound( gafferWindow, boundData ) :
+
+	window = gafferWindow._qtWidget().windowHandle()
+	if not window :
+		return
+
+	# If we don't clear the full-screen flag then restoring geometry won't
+	# work, as its overridden by the full-screen state.  Doing it conditionally
+	# improves fullscreen - fullscreen layout transitions (avoids pointless
+	# re-scale animations), but does mean that we don't properly restore the
+	# non-fullscreen size of the window and it will keep that of the first
+	# fullscreen layout loaded. Seems a reasonable trade-off though.
+	if not ( boundData["fullScreen"] and window.windowState() == QtCore.Qt.WindowFullScreen ) :
+		window.setWindowState( QtCore.Qt.WindowNoState )
+
+	targetScreen = QtWidgets.QApplication.primaryScreen()
+
+	if boundData["screen"] > -1 :
+		screens = QtWidgets.QApplication.screens()
+		if boundData["screen"] < len(screens) :
+			targetScreen = screens[ boundData["screen"] ]
+			window.setScreen( targetScreen )
+
+	screenGeom = targetScreen.availableGeometry()
+	window.setGeometry(
+		( boundData["bound"].min()[0] * screenGeom.width() ) + screenGeom.x(),
+		( ( 1.0 - boundData["bound"].max()[1] ) * screenGeom.height() ) + screenGeom.y(),
+		( boundData["bound"].size()[0] * screenGeom.width() ),
+		( boundData["bound"].size()[1] * screenGeom.height() )
+	)
+
+	if boundData["fullScreen"] :
+		window.setWindowState( QtCore.Qt.WindowFullScreen )
+	elif boundData["maximized"] and sys.platform != "darwin" :
+		window.setWindowState( QtCore.Qt.WindowMaximized )
+	else :
+		window.setWindowState( QtCore.Qt.WindowNoState )
+
+def _serialiseSplits( splitContainer, scriptNode = None ) :
```

Comment:
Should this be a method of `_SplitContainer()`, and should `_SplitContainer().addChildren()` be renamed to make it clear that it is for deserialisation of the result? At the moment, this relationship is pretty unclear because the functions are in totally different places, and addChildren was badly named from before.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3167/commits/10d5c6b413c2e765e03553a062a80929e1921e5a